cmd/gateway/commands_test.go

```diff
@@ -391,3 +393,134 @@ func TestSleepCmdFlagValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestParseFlagKeysAndValues(t *testing.T) {
+	g := NewWithT(t)
+
+	flagSet := pflag.NewFlagSet("flagSet", 0)
+	// set SortFlags to false for testing purposes so when parseFlagKeysAndValues loops over the flagSet it
+	// goes off of primordial order.
+	flagSet.SortFlags = false
+
+	var boolFlagTrue bool
+	flagSet.BoolVar(
+		&boolFlagTrue,
+		"boolFlagTrue",
+		true,
+		"boolean true test flag",
+	)
+
+	var boolFlagFalse bool
+	flagSet.BoolVar(
+		&boolFlagFalse,
+		"boolFlagFalse",
+		false,
+		"boolean false test flag",
+	)
+
+	intFlagDefault := intValidatingValue{
+		validator: validatePort,
+		value:     8080,
+	}
+	flagSet.Var(
+		&intFlagDefault,
+		"intFlagDefault",
+		"default int test flag",
+	)
+
+	intFlagUserDefined := intValidatingValue{
+		validator: validatePort,
+		value:     8080,
+	}
+	flagSet.Var(
+		&intFlagUserDefined,
+		"intFlagUserDefined",
+		"user defined int test flag",
+	)
+	err := flagSet.Set("intFlagUserDefined", "8081")
+	g.Expect(err).To(Not(HaveOccurred()))
+
+	stringFlagDefault := stringValidatingValue{
+		validator: validateResourceName,
+		value:     "default-string-test-flag",
+	}
+	flagSet.Var(
+		&stringFlagDefault,
+		"stringFlagDefault",
+		"default string test flag",
+	)
+
+	stringFlagUserDefined := stringValidatingValue{
+		validator: validateResourceName,
+		value:     "user-defined-string-test-flag",
+	}
+	flagSet.Var(
+		&stringFlagUserDefined,
+		"stringFlagUserDefined",
+		"user defined string test flag",
+	)
+	err = flagSet.Set("stringFlagUserDefined", "changed-test-flag-value")
+	g.Expect(err).To(Not(HaveOccurred()))
+
+	namespacedNameFlagDefault := namespacedNameValue{
```

Comment:
this flag is just a custom string flag that we already covered above

However, what we don't cover is when the value is empty. So let's not set the default value in the value field.

So let's just have customStringFlagNoDefaultValueUnset and customStringFlagNoDefaultValueUserDefined.

Note that in #1544 we're adding custom string flags without default value, where those flags are not required:
```
		usageReportClusterName = stringValidatingValue{
			validator: validateQualifiedName,
		}
		usageReportSecretName = namespacedNameValue{}
		usageReportServerURL  = stringValidatingValue{
			validator: validateURL,
		}
```
so such two cases will match those things

Reply:
I think I resolved this, but unsure